### PR TITLE
[terra-icon MVB] added upgrade guide

### PR DESCRIPTION
**Pull request by sdadn**
### Summary

Added an upgrade guide for the MVB for terra-icon

Closes # UXPLATFORM-6554

### Deployment Link

https://terra-core-deployed-pr-#.herokuapp.com/

### Additional Details

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-core-docs/src/terra-dev-site/doc/icon/UpgradeGuide.5.doc.mdx [hunk 1]
```diff
@@ -4,7 +4,30 @@ import { Badge } from 'terra-icon/package.json?dev-site-package';
...
-## Changes from 2.0 to 3.0
+- [Upgrading from 3.0 to 4.0](#upgrading-from-3.0-to-4.0)
+- [Upgrading from 2.0 to 3.0](#upgrading-from-2.0-to-3.0)
+
+## Upgrading from 3.0 to 4.0
+
+- Icons are now split into `meaningful icons` and `decorative icons`. By default, icons are imported as `meaningful`. To import `decorative` icons, the import statement must be updated:
+
+```js
+// importing a meaninful icon
```
Comment (saket2403) [inline packages/terra-core-docs/src/terra-dev-site/doc/icon/UpgradeGuide.5.doc.mdx:15]: Typo- Meaningful

Comment (xenoworf) [inline packages/terra-core-docs/src/terra-dev-site/doc/icon/UpgradeGuide.5.doc.mdx:15]:
```suggestion
// importing a meanginful icon
```

File: packages/terra-core-docs/src/terra-dev-site/doc/icon/UpgradeGuide.5.doc.mdx [hunk 2]
```diff
@@ -4,7 +4,30 @@ import { Badge } from 'terra-icon/package.json?dev-site-package';
...
+import Star           from "terra-icon/lib/icon/IconFeaturedOutlineYellow";
```
Comment (xenoworf) [inline packages/terra-core-docs/src/terra-dev-site/doc/icon/UpgradeGuide.5.doc.mdx:16]:
```suggestion
import Star from "terra-icon/lib/icon/IconFeaturedOutlineYellow";
```

**Review by xenoworf — APPROVED** [review]

File: packages/terra-core-docs/CHANGELOG.md [hunk 3]
```diff
@@ -2,29 +2,23 @@
```
Comment (sdadn) [inline packages/terra-core-docs/CHANGELOG.md:2]: Note: There were some errors/duplicates in the CHANGELOG caused by merge conflicts in a previous commit. This PR corrects those errors and the CHANGELOG should now be accurate.

File: packages/terra-core-docs/src/terra-dev-site/doc/icon/UpgradeGuide.5.doc.mdx (hunk 1 above)
Comment (sdadn) [reply packages/terra-core-docs/src/terra-dev-site/doc/icon/UpgradeGuide.5.doc.mdx]: Corrected here: 86e3649ef08fe9ba45ef51f7ed7a4f4fccc3c36e

Comment (sdadn) [reply packages/terra-core-docs/src/terra-dev-site/doc/icon/UpgradeGuide.5.doc.mdx]: Corrected here: 86e3649ef08fe9ba45ef51f7ed7a4f4fccc3c36e

**Review by saket2403 — APPROVED** [review]

**Comment by sdadn** [comment]
Merged the PR even though the build is failing since this PR is essentially just a documentation update and the failures are caused by other underlying issues. These will be resolved in a separate PR.

**Pull request merged**